src/commands/cmd_query.c

```diff
@@ -104,21 +90,21 @@ void Graph_Query(void *args) {
 
 	// Set policy after lock acquisition, avoid resetting policies between readers and writers.
 	Graph_SetMatrixPolicy(gc->g, SYNC_AND_MINIMIZE_SPACE);
-
+	result_set = NewResultSet(ctx, compact);
+	QueryCtx_SetResultSet(result_set);
```

Comment:
As a general note (not for this PR), we should refactor signatures to take advantage of this soon. Things like `ExecutionPlan_Execute` returning a `ResultSet`, especially, are confusing now!

Additionally, I don't think PROFILE commands work properly regardless right now, but these changes should be mirrored in `cmd_profile.c`.